Automattic/jetpack
# Editors: Add first post published modal

Fixes https://github.com/Automattic/wp-calypso/issues/72803

## Estimated Time to Test / Review:
- Test -> Med / Long
- Review -> Med

## Proposed changes:


 There are multiple modals that want to respond to the post editor's publish / save / update button. To address this, we're consolidating all modal logic into the Jetpack repo. This has the added benefit of ensuring that the first post published modal will always appear, because, while the user can disable the editing toolkit plugin, they cannot disable the jetpack features we're implementing

* Consolidate first post published modal logic [that was removed from Calypso](https://github.com/Automattic/wp-calypso/pull/73340) into Jetpack

## GIFS

NOTE: This is a first iteration. We can match the original first post published modal content easily in a follow-up PR.

![2023-02-14 11 51 29](https://user-images.githubusercontent.com/5414230/218858728-c3045999-8a42-42b6-b25f-834c3fe072d0.gif)

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






In order for this PR to behave as expected, **we'll need to sync these ETK changes to our sandbox first** https://github.com/Automattic/wp-calypso/pull/73340. Otherwise, logic from the original modal will interfere with the duplicated logic here.

* Open Calypso repo in a terminal
* Check out `remove/post-published-modal` https://github.com/Automattic/wp-calypso/pull/73340
* [Create a new site through general onboarding](wordpress.com/start)
* Select `Write & Publish` as the site's goal
* Select any category
* Use any site title and tagline
* Before moving forward, sandbox your new site ( Don't forget to reset the DNS cache by forcing reconnection to autoproxxy or other means )
* Navigate to `apps/editing-toolkit` in a terminal
*  `yarn dev --sync`

Now we sync our jetpack changes

* Open Jetpack repo in a terminal
* Check out this branch
* Ensure that the local dev environment is working properly ( `pnpm install`, `pnpm jetpack install`, etc. )
* `pnpm jetpack watch plugins/jetpack --production`
* In another terminal `pnpm jetpack rsync`
  * When prompted, select `jetpack` 
  * If you already have an alias for destination host to sync the Jetpack plugin, use that. Otherwise, select `Create new`
  * Input destination host Ex. `wpdev@YOUR_SANDBOX_URL:~/public_html/wp-content/mu-plugins/jetpack-plugin/production`. Your sandbox url can be found in your ~/.ssh/config file if you have one. If you still can't find it, feel free to reach out.
  * Documentation at pdWQjU-6L-p2
* Continue onboarding by clicking on the `Skip to dashboard` button
* When you land in the Launchpad, click on the `Edit site design` task
* Make changes in the site editor and save them
* VERIFY that the launchpad modal continues to behave as expected
* Navigate back to the Launchpad by clicking on the site editor back to dashboard button
* Click on the `Write your first post` task
* Edit the post's content
* Click on the publish button
* VERIFY that only one modal is shown
* VERIFY that the modal displays first post published text
* VERIFY that the `View post` button navigates to the post in the frontend
* VERIFY that subsequent saves in the post editor display the launchpad modal
